compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -1354,7 +1356,31 @@ impl<'a> Linker for WasmLd<'a> {
     fn subsystem(&mut self, _subsystem: &str) {}
 
     fn linker_plugin_lto(&mut self) {
-        // Do nothing for now
+        match self.sess.opts.cg.linker_plugin_lto {
+            LinkerPluginLto::Disabled => {
+                // Nothing to do
+            }
+            LinkerPluginLto::LinkerPluginAuto => {
```

Comment:
```suggestion
            LinkerPluginLto::LinkerPluginAuto | LinkerPluginLto::LinkerPlugin(_) => {
```
`push_linker_plugin_lto_args` could then be inlined and removed.

Upd: the match on `self.sess.opts.optimize` could be kept separate function, though, it is shared between `fn optimize` and `fn linker_plugin_lto`.

Reply:
I intentionally left them separate.

- That the match turns out the same is more of a coincidence than anything else. I anticipate the  `--lto-O*` args may change to support Os and Oz in future. (Especially with those being the most popular opt levels for webassembly.) If that happens, adding Os/Oz will break the normal `fn optimize` that controls merging sections only, unrelated to LLVM.
- `push_linker_plugin_lto_args` could be inlined, but I was going to do another PR that puts warnings in one of the match arms but not the other.